quickwit-cli/src/template.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright (C) 2022 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashMap;
+
+use anyhow::{bail, Context, Result};
+use new_string_template::template::Template;
+use once_cell::sync::Lazy;
+use quickwit_storage::OwnedBytes;
+use regex::Regex;
+use tracing::{debug, warn};
+
+// Matches ${value} if value is in format of:
```

Comment:
I am unsure if we should support `:+` since it would probably raise an error to deserialize empty strings. 

We could do something like removing the field from config if `VAR` is not set to a value. That way, it would fall back to its default value.

Reply:
Ok, let's not support `:+`.